extensions/functions_arithmetic.yaml

```diff
@@ -181,6 +211,72 @@ scalar_functions:
           - value: i64
           - value: i64
         return: i64
+  -
+    name: "power"
+    description: >
+      Take the power with the first value as the base and second as exponent.
+
+      All integer arguments are promoted to int64.
+      All floating point arguments are promoted to float64.
+    impls:
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i8
+          - value: i8
+        return: i64
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i16
+          - value: i16
+        return: i64
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i32
+          - value: i32
+        return: i64
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i64
+          - value: i64
+        return: i64
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: fp32
+          - value: fp32
+        return: fp64
```

Comment:
I don't really see why fp32 ^ fp32 should promote to fp64. Both have "similar" ranges in that they include infinity, and if you really need double accuracy you should just promote to double first. If we take the C math library as precedent, we have `double pow(double, double)` and float `powf(float, float)`.

I feel the same about the integer power functions, though a bit less strongly I guess. Why should power be different from multiplication and addition? Both can overflow, and the former requires double the bits. All languages I know (except VHDL/Verilog, but those are kinda special) require promotion first if you need a larger output type. I guess I can see where it's coming from, but I don't think this sort of smartness belongs in the Substrait spec.

IMO: just let the producer decide if they want to upcast or not, and if a compound operation that does the upcast and the operation simultaneously exists in the consumer it is up to them to optimize to that operation when they encounter the `pow(cast<i64>(x), cast<i64>(x))` pattern.